docs/src/pages/docs/functions/helpers.md

```diff
@@ -283,6 +283,66 @@ that can be called in any combination, such as: `f(x, y, z)`, `f(x)(y)(z)`,
 `f(x, y)(z)`, or even `f(x)(y, z)`. This is great for doing partial application
 on functions for maximum re-usability.
 
+```javascript
+import curry from 'crocks/helpers/curry'
+
+// add :: (Number, Number, Number) -> Number
+const add = (a, b, c) =>
+  a + b + c
+
+// partial application impossible
+add(1)
+// => 1 + undefined + undefined => NaN
+
+// NOT the only possible type signature as highlighted below
+// curriedAdd :: Number -> Number -> Number -> Number
+const curriedAdd = curry( add )
+
+curriedAdd(1)
+// => [Function: curried]
+
+// Number -> Number -> Number -> Number
+curriedAdd(1)(2)(3)
+// => 6
+
+// (Number, Number) -> Number -> Number
+curriedAdd(1, 2)(3)
+// => 6
+
+// Number -> (Number -> Number) -> Number
+curriedAdd(1)(2, 3)
+// => 6
+
+// (Number, Number, Number) -> Number
+curriedAdd(1, 2, 3)
+// => 6
+
+// optionalAdd :: (Number -> Number -> Number) -> Number
+const optionalAdd = (a, b, c = 2) =>
```

Comment:
I am thinking this is too much information for this. example, maybe just add something to the paragraph above, warning of this case. Seems to cloud the intent of the function here.

Reply:
This was added in response to a [comment](https://github.com/evilsoft/crocks/issues/454#issuecomment-575936386) made by @dalefrancis88. I'll wait on clarification, it may be that I didn't express his idea well.

Reply:
Gotcha. Yeah, I think it is good to mention like @dalefrancis88 said.
I think we should frame it more as a warning not to use default args, but it is hard to explain why without launching into some white board discussion on how Function Inputs can be considered Product Types and how Products are nothing more than Limits, and because JS does not use Parametric Polymorphism, we cannot accurately represent a Limit without implementing a case for EVERY single type in JS.

So short of that maybe we can show how to deal with an existing function/method that uses argument defaults instead of showing them creating one from scratch. Kinda steer them away from the idea. And mention that they should be avoid them, without a real reason, other than it is hard to use curry with them.

Then if people ask about it, we can explain it in more depth in future issues or chat room discussions.

Reply:
OK. Off the top of my head, I tested `curry`-ing `console.log` and `JSON.stringify` and the results were consistent with my understanding for `console.log`, but strange for `JSON.stringify`.

```
const curriedConsolelog =
  curry(console.log)

curriedConsolelog("Hello ", "me Matey!")
//=> Hello
// As expected, the optional arguments after the first are ignored

const curriedStringify =
  curry(JSON.stringify)

curriedStringify({'head': 1})
//=> [Function curried]
// Unexpected. Apparently JSON.stringify uses another mechanism for optional parameters

curriedStringify({'head': 1}, null, '')
//=> {'head': 1}

curriedStringify({'head': 1})(null)('')
//=> {'head': 1}
```

Reply:
Looking at the code for `JSON.stringify` @ [crockford's](https://github.com/douglascrockford/JSON-js/blob/master/json2.js) the `stringify` method doesn't use optional parameters as such. It allows them to be `undefined` and recovers with checks and balances.

Reply:
Apologies for the confusion, my comment was more as @evilsoft said, just a bit of a warning so that it sits in peoples subconscious for when they accidentally run into the issue. 

We've run into that issue with `stringify` and `parse` before, it's quite annoying. The way we get round that is to wrap it with `unary` and then it passes `undefined` to the fake "optional" parameters